frontend/benefit/applicant/src/hooks/useFormActions.tsx

```diff
@@ -29,21 +29,19 @@ const useFormActions = (application: Application): FormActions => {
     application.applicationStep ?? ''
   );
 
-  const {
-    mutateAsync: createApplication,
-    data: newApplication,
-    error: createApplicationError,
-  } = useCreateApplicationQuery();
-
-  useEffect(() => {
-    if (newApplication?.id) {
-      void router.replace({
-        query: {
-          id: newApplication.id,
-        },
-      });
-    }
-  }, [newApplication?.id, router]);
+  const { mutateAsync: createApplication, error: createApplicationError } =
+    useCreateApplicationQuery();
+
+  const createApplicationAndAppendId = async (
+    data: ApplicationData
+  ): Promise<void> => {
+    const newApplication = await createApplication(data);
+    void router.replace({
```

Comment:
router replace causes page refresh, but you can prevent it by passing `{ shallow: true }`. I'm not sure though which is the use case here.

Reply:
Good point 👍 
Fixed